servicebus/data-plane/azure-servicebus/src/main/java/com/microsoft/azure/servicebus/primitives/RequestResponseLink.java

```diff
@@ -861,18 +861,30 @@ private void runSendLoop() {
                             this.parent.exceptionallyCompleteRequest((String)requestToBeSent.getRequest().getMessageId(), new PayloadSizeExceededException(String.format("Size of the payload exceeded Maximum message size: %s kb", this.maxMessageSize / 1024), exception), false);
                         }
 
-                        try {
-                            int sentMsgSize = this.sendLink.send(encodedPair.getFirstItem(), 0, encodedPair.getSecondItem());
-                            assert sentMsgSize == encodedPair.getSecondItem() : "Contract of the ProtonJ library for Sender.Send API changed";
-                            delivery.settle();
-                            this.availableCredit.decrementAndGet();
-                            TRACE_LOGGER.debug("RequestResonseLink {} internal sender sent a request. available credit :{}", this.parent.linkPath, this.availableCredit.get());
-                        } catch(Exception e) {
-                            TRACE_LOGGER.error("RequestResonseLink {} failed to send request with request id:{}.", this.parent.linkPath, requestIdToBeSent, e);
-                            this.parent.exceptionallyCompleteRequest(requestIdToBeSent, e, false);
+                        if (encodedPair != null) {
```

Comment:
[SpotBugs-P2]
Possible null pointer dereference of encodedPair in com.microsoft.azure.servicebus.primitives.RequestResponseLink$InternalSender.runSendLoop() on exception path

A reference value which is null on some exception control path is dereferenced here.  This may lead to a NullPointerException when the code is executed.  Note that because SpotBugs currently does not prune infeasible exception paths, this may be a false warning.
Also note that SpotBugs considers the default case of a switch statement to be an exception path, since the default case is often infeasible.